subroutines/habits/tasks.py

```diff
@@ -0,0 +1,125 @@
+from datetime import timedelta
+
+from django.db.models import Exists, OuterRef
+from django.utils import timezone
+
+from config import celery_app
+from subroutines.users.models import User
+from subroutines.habits.models import Habit, Instance
+
+
+@celery_app.task(max_retries=3)
+def create_habit_instances(habit_pk):
+    """
+    Each time a habit is created this task is called
+    to create instances.
+    """
```

Comment:
Creating habits in advance shouldn't happen. That was one of the main requirements. You should only creare habits for the current day.